packages/financial-templates-lib/src/price-feed/UniswapPriceFeed.js

```diff
@@ -86,14 +82,40 @@ class UniswapPriceFeed extends PriceFeedInterface {
 
   // Not part of the price feed interface. Can be used to pull the uniswap price at the most recent block.
   getLastBlockPrice() {
-    return this.lastBlockPrice && this.convertPoolDecimalsToPriceFeedDecimals(this.lastBlockPrice);
+    return this.lastBlockPrice && this.convertBaseDecimalsToPriceFeedDecimals(this.lastBlockPrice);
   }
 
   getPriceFeedDecimals() {
     return this.priceFeedDecimals;
   }
 
   async update() {
+    // Read token0 and token1 precision from Uniswap contract.
+    const [token0Address, token1Address] = await Promise.all([
```

Comment:
Since we don't expect token0 and token1 to change precisions, could we cache the results by doing something like:
```js
if (!this.quoteCurrencyDecimals || !this.convertBaseDecimalsToPriceFeedDecimals) {
  // Logic to set this.quoteCurrencyDecimals and this.convertBaseDecimalsToPriceFeedDecimals
}

// Grab price info from uniswap contract.
```

Reply:
Yeah good point. I'll implement